MathPortal/MathPortal/Source/UserInterface/TabBarViewControllers/SolvedTasks/SolveTask/SolveTaskViewController.swift

```diff
@@ -28,17 +28,63 @@ class SolveTaskViewController: UIViewController {
             controller?.task = task
             return controller
         }(), animationStyle: .fade)
+        Appearence.setUpnavigationBar(controller: self, leftBarButtonTitle: "< Back", leftBarButtonAction: #selector(goBack), rightBarButtonTitle: "Save", rightBarButtonAction: #selector(save))
         taskContentController?.layer.cornerRadius = 10
         equationsTableView?.layer.cornerRadius = 10
         equationsTableView?.register(R.nib.taskViewControllerTableViewCell)
         equationsTableView?.customDelegate = self
+        reloadSolution()
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        guard equations.count < 0 else { return }
```

Comment:
Never guard methods unrelated properties. This should be 

```
if !equations.isEmpty {
    equationsTableView?.scrollToRow(at: IndexPath(row: equations.count - 1, section: 0), at: .middle, animated: false)
}
```

In general it is OK to guard in cases like
`func doSomethingWithA() { guard a = a else { return } }`
And not
`func doSomethingWithA() { guard b = b else { return } }`

The reason for this is that methods like this one can get extra unrelated logic and `guard` will remove that logic as well. It is easy for developer to miss such things... Imagine:

```
override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        guard equations.count < 0 else { return }
        equationsTableView?.scrollToRow(at: IndexPath(row: equations.count - 1, section: 0), at: .middle, animated: false)

        someButton.animateHidden(false)
}
```

The last line is completely unrelated to the first part but still makes complete sense to be put into this method. But it will not be executed when `equations` is empty which is a bug.